packages/async-rewriter2/README.md

```diff
@@ -63,7 +63,62 @@ function foo() {
 Note how identifiers remain accessible in the outside environment, including
 top-level functions being hoisted to the outside.
 
-### Step two: Async function wrapping
+### Step two: Making certain exceptions uncatchable
+
+In order to support Ctrl+C properly, we add a type of exception that is not
+catchable by userland code.
+
+For example,
+
+```js
+try {
+  foo3();
+} catch {
+  bar3();
+}
+```
+
+is transformed into
+
+```js
+try {
+  foo3();
+} catch (_err) {
+  if (!err || !_err[Symbol.for('@@mongosh.uncatchable')]) {
+    bar3();
+  }
```

Comment:
I think for the sake of correctness there should be the throw in the else branch
```suggestion
  } else {
    throw _err;
  }
```